libraries/AP_ADSB/sagetech-sdk/sgUtil.h

```diff
@@ -0,0 +1,325 @@
+/**
+ * @copyright Copyright (c) 2021 Sagetech, Inc. All rights reserved.
+ *
+ * @file   sgUtil.h
+ * @author jimb
+ *
+ * @date Feb 2, 2021
+ */
+
+#ifndef UTIL_H
+#define UTIL_H
+
+#include <stdint.h>
+
+#ifndef swap16
+#define swap16(data) \
+    ((((data) >> 8) & 0x00FF) | (((data) << 8) & 0xFF00))
+#endif
+
+#ifndef swap24
+#define swap24(data) \
+    (((data) >> 16) | ((data)&0x00FF00) | (((data) << 16) & 0xFF0000))
+#endif
+
+#ifndef swap32
+#define swap32(data) \
+    (((data) >> 24) | (((data)&0x00FF0000) >> 8) | (((data)&0x0000FF00) << 8) | ((data) << 24))
+#endif
+
+#ifndef swap64
+#define swap64(data) \
+   (swap32((data & 0x00000000ffffffffULL))) << 32 | swap32(data >> 32))
+#endif
+
+#ifndef PI
+#define PI 3.14159265359
+#endif
```

Comment:
No reason to define your own PI if there's already standard libraries for that.

```suggestion
#ifndef M_PI
#define M_PI 3.14159265359
#endif
```

Reply:
PI and not M_PI is used in our SDK somewhere, and we didn't use all standard libraries for certification purposes. This shouldn't affect any other definitions of PI in ardupilot due to the infndef.